chips/rp2040/src/pio.rs

```diff
@@ -506,6 +515,52 @@ const PIO1_SET_BASE: StaticRef<PioRegisters> =
 const PIO1_CLEAR_BASE: StaticRef<PioRegisters> =
     unsafe { StaticRef::new((PIO_1_BASE_ADDRESS + 0x3000) as *const PioRegisters) };
 
+/// Used to relocate program in the PIO memory.
+pub struct RelocatedProgram<'a, I>
```

Comment:
Please add a little more documentation about what this does and support to with links from the RP2040's Datasheet?

Reply:
The datasheet is available [here](https://datasheets.raspberrypi.com/rp2040/rp2040-datasheet.pdf)
Section 3 describes PIO usage, 3.3 defines pio_asm